**Q1. Write a function that stutters a word as if someone is
struggling to read it. The first two letters are repeated twice with an
ellipsis ... and space after each, and then the word is pronounced with
a question mark ?.**

**Examples**

stutter("incredible") ➞ "in... in... incredible?"

stutter("enthusiastic") ➞ "en... en... enthusiastic?"

stutter("outstanding") ➞ "ou... ou... outstanding?"

Hint :- Assume all input is in lower case and at least two characters
long.

In [58]:
def stutter(string):
    print(string[0:2],'...',string[0:2],'...',string )   
        
stutter('enthusuastic')        

en ... en ... enthusuastic


**Q2 Create a function that takes an angle in radians and returns
the corresponding angle in degrees rounded to one decimal place.**

### Examples

radians\_to\_degrees(1) ➞ 57.3

radians\_to\_degrees(20) ➞ 1145.9

radians\_to\_degrees(50) ➞ 2864.8

In [74]:
def convert_to_deg():
    angle = float(input('Enter the Angle in radians\n'))
    print('The angle',angle,'entered in radians, converted in degrees is',round(angle*57.3,1))
convert_to_deg()    
    

Enter the Angle in radians
23.4334
The angle 23.4334 entered in radians, converted in degrees is 1342.7


**Q3. In this challenge, establish if a given integer num is a
Curzon number. If 1 plus 2 elevated to num is exactly divisible by 1
plus 2 multiplied by num, then num is a Curzon number.**

**Given a non-negative integer num, implement a function that returns True
if num is a Curzon number, or False otherwise.**

### Examples

is\_curzon(5) ➞ True

\# 2 \*\* 5 + 1 = 33

\# 2 \* 5 + 1 = 11

\# 33 is a multiple of 11

is\_curzon(10) ➞ False

\# 2 \*\* 10 + 1 = 1025

\# 2 \* 10 + 1 = 21

\# 1025 is not a multiple of 21

is\_curzon(14) ➞ True

\# 2 \*\* 14 + 1 = 16385

\# 2 \* 14 + 1 = 29

\# 16385 is a multiple of 29


In [79]:
def is_curzon():
    num = int(input('Enter the number\n\n'))
    a = 2**num + 1
    b = 2*num + 1
    if a%b==0:
        print(num,'is a curzon number')
    else :
        print(num,'is not a curzon number')
is_curzon()        

Enter the number

5
5 is a curzon number


In [81]:
is_curzon()        

Enter the number

10
10 is not a curzon number


In [83]:
is_curzon()        

Enter the number

14
14 is a curzon number


**Q4.Given the side length x find the area of a hexagon.**

<img src="attachment:media/image1.jpeg" style="width:4.79167in;height:3.59167in" alt="Formula to find the area of a hexagon" />

### Examples

area\_of\_hexagon(1) ➞ 2.6

area\_of\_hexagon(2) ➞ 10.4

area\_of\_hexagon(3) ➞ 23.4



In [89]:
def area_of_hexagon():
    num = int(input('Enter the side length'))
    print('The are of the Hexagon is',(3*(3**(0.5))*(num**2))/2)
area_of_hexagon()    

Enter the side length44
The are of the Hexagon is 5029.87554518002


**Q5. Create a function that returns a base-2 (binary)
representation of a base-10 (decimal) string number. To convert is
simple: ((2) means base-2 and (10) means base-10) 010101001(2) = 1 + 8 +
32 + 128.**

**Going from right to left, the value of the most right bit is 1, now from
that every bit to the left will be x2 the value, value of an 8 bit
binary numbers are (256, 128, 64, 32, 16, 8, 4, 2, 1).**

### Examples

binary(1) ➞ "1"

\# 1\*1 = 1

binary(5) ➞ "101"

\# 1\*1 + 1\*4 = 5

binary(10) ➞ "1010"

\# 1\*2 + 1\*8 = 10

In [90]:
import math

In [93]:
def binary(decimal):
    result = ''
    if (decimal == 0):
        result = '0'
    while (decimal > 0):
        result += result % 2
        decimal = math.floor(decimal/2)
    return result.split("").reverse().join("")
binary(1)

TypeError: not all arguments converted during string formatting